Configuration Caching is not allowing to publish* from main

Enabling configuration caching broke the Detekt release process. I.e. this PR
- https://github.com/detekt/detekt/pull/4576

I had to set `org.gradle.unsafe.configuration-cache=false`, otherwise I could not publish artifacts from this repo.

## Expected Behavior
You should be able to run `./gradlew publishToMavenLocal` from this repo, and it should be green.

## Observed Behavior
Running `./gradlew publishToMavenLocal` fails with:

```
Configuration cache is an incubating feature.
Calculating task graph as no configuration cache is available for tasks: publishToMavenLocal
Type-safe dependency accessors is an incubating feature.
Type-safe project accessors is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

1313 problems were found storing the configuration cache, 262 of which seem unique.
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.tasks.DefaultTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.artifact.AbstractMavenArtifact$CompositeTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.artifact.SingleOutputTaskMavenArtifact$GeneratorTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication', a subtype of 'org.gradle.api.publish.Publication', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.tasks.GenerateMavenPom', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:task_access
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.tasks.GenerateModuleMetadata', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:task_access
- Task `:custom-checks:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.tasks.bundling.Jar', a subtype of 'org.gradle.api.Task', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:task_access
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.internal.tasks.DefaultTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.artifact.AbstractMavenArtifact$CompositeTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.artifact.SingleOutputTaskMavenArtifact$GeneratorTaskDependency', a subtype of 'org.gradle.api.tasks.TaskDependency', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:detekt-api:signDetektPublicationPublication` of type `org.gradle.plugins.signing.Sign`: cannot serialize object of type 'org.gradle.api.publish.maven.internal.publication.DefaultMavenPublication', a subtype of 'org.gradle.api.publish.Publication', as these are not supported with the configuration cache.
  See https://docs.gradle.org/7.4/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
plus 247 more problems. Please see the report for details.

See the complete report at file:///Users/ncor/oss/detekt/build/reports/configuration-cache/exkjnx4luxnzjkqnq477xpzx4/6xwz9dli9g6dhpt446vaptrr3/configuration-cache-report.html

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 5s
292 actionable tasks: 93 executed, 5 from cache, 194 up-to-date
Configuration cache entry discarded with 1313 problems.
```

## Context
We should setup a CI job that tries to `publishToMavenLocal`.

--- a/build-logic/src/main/kotlin/packaging.gradle.kts
+++ b/build-logic/src/main/kotlin/packaging.gradle.kts
@@ -11,6 +11,10 @@ val sonatypePassword: String? = findProperty("sonatypePassword")
     ?.toString()
     ?: System.getenv("MAVEN_CENTRAL_PW")
 
+tasks.withType<Sign>().configureEach {
+    notCompatibleWithConfigurationCache("https://github.com/gradle/gradle/issues/13470")
+}
+
 publishing {
     repositories {
         maven {

